File: Templates/Java/integrationtesting/src/main/java/com/example/integrationtesting/UserController.java
package com.example.integrationtesting;

import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Optional;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;

@RestController
public class UserController {

    private final UserService userService;
    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("/users")
    public List<User> getAllUsers() {
        return userService.getAll();
    }

    @GetMapping("/user/{id}")
    public Optional<User> getUserById(@PathVariable String id) {
        return userService.getById(id);
    }
    
    @GetMapping("/users/nearby")
    public List<AddressUser> getNearby(@RequestParam("lat") double lat, @RequestParam("lng") double lng, @RequestParam("miles") double miles) {
        return userService.getNearby(lat, lng, miles);
    }
}

File: Templates/Kotlin/integrationtesting/src/main/java/com/example/integrationtesting/UserController.kt
package com.example.integrationtesting

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController(private val userService: UserService) {
    @get:GetMapping("/users")
    val allUsers: MutableList<User?>?
        get() = userService.all

    @GetMapping("/user/{id}")
    fun getUserById(@PathVariable id: String): User? {
        return userService.getById(id)
    }

    @GetMapping("/users/nearby")
    fun getNearby(
        @RequestParam("lat") lat: Double,
        @RequestParam("lng") lng: Double,
        @RequestParam("miles") miles: Double
    ): MutableList<AddressUser?>? {
        return userService.getNearby(lat, lng, miles)
    }
}
